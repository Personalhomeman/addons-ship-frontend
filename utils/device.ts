import * as ReactDeviceDetect from 'react-device-detect';
import * as CompareVersions from 'compare-versions';

export const isAndroid = () => ReactDeviceDetect.isAndroid;
export const isIOS = () => ReactDeviceDetect.isIOS;
export const osVersion = () => ReactDeviceDetect.osVersion;
export const mobileModel = () => ReactDeviceDetect.mobileModel;
export const compareVersions = (firstVersion: string, secondVersion: string) => CompareVersions(firstVersion, secondVersion)
