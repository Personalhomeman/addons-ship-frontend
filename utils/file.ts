import fetch from 'node-fetch';

export const uploadFileToS3 = async (file: File | undefined, presignedUrl: string | undefined) => {
  if (!file || !presignedUrl) throw new Error('File or presignedUrl is invalid');

  const options = {
    method: 'put',
    body: file as any
  };

  return fetch(presignedUrl, options).then(resp => {
    if (!resp.ok) throw new Error(`${resp.status}: ${resp.statusText}`);
  });
};
