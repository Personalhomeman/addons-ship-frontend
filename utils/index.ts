import mapKeys from 'lodash/mapKeys';
import mapValues from 'lodash/mapValues';
import camelCase from 'lodash/camelCase';
import snakeCase from 'lodash/snakeCase';
import isPlainObject from 'lodash/isPlainObject';

export const actionTypeCreator = (prefix: string) => (actionType: TemplateStringsArray) => `${prefix}_${actionType}`;

export const camelizeKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => camelCase(key)) as unknown) as T;

export const camelizeKeysDeep = <T>(input: Object): T =>
  mapValues(camelizeKeys(input), (value: any) => (isPlainObject(value) ? camelizeKeysDeep(value) : value)) as T;

export const snakifyKeys = <T>(input: Object): T =>
  (mapKeys(input, (_, key: string) => snakeCase(key)) as unknown) as T;

export const snakifyKeysDeep = <T>(input: Object): T =>
  mapValues(snakifyKeys(input), (value: any) => (isPlainObject(value) ? snakifyKeysDeep(value) : value)) as T;
