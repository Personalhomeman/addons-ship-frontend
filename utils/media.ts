import { useState, useEffect } from 'react';

type Handler = (e: MediaQueryListEvent) => void;

const getListeners = (queries: string[], handler: Handler) =>
  queries.map(query => {
    const listener = window.matchMedia(query);

    listener.addListener(handler);

    return listener;
  });

export const mediaQuery = (...rules: string[]) => {
  const mediaRules = rules.map(r => `(min-width: ${r})`);

  const stateHandlers = rules.map(() => useState(false));

  const getStateHandler = (media: string) => {
    const idx = mediaRules.indexOf(media);

    return stateHandlers[idx];
  };

  useEffect(() => {
    const handler = ({ matches, media }: MediaQueryListEvent) => {
      const [, set] = getStateHandler(media);

      set(matches);
    };

    const listeners = getListeners(mediaRules, handler);

    listeners.forEach(({ media, matches }) => {
      const [, set] = getStateHandler(media);
      set(matches);
    });

    return () => listeners.forEach(l => l.removeListener(handler));
  });

  return stateHandlers.map(([isMatch]) => isMatch);
};
