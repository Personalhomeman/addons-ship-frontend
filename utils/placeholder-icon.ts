import { placeholderIcons } from '@/config';
import { ProjectType } from '@/models/app';

export const getIcon = (appIcon?: string, projectType?: ProjectType) => {
  let icon = appIcon;

  if (!icon) {
    icon = placeholderIcons[projectType!] || placeholderIcons.other;
  }

  const suffix = Math.random()
    .toString(16)
    .replace(/^0\./, '');

  return `${icon}?${suffix}`;
};
