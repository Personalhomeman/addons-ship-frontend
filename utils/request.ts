import fetch, { RequestInit } from 'node-fetch';

const getHeaders = (token: string) => ({
  Authorization: token
});

interface RequestParams extends RequestInit {
  url: string;
  token: string;
}

export const request = ({ url, method, token, body }: RequestParams) =>
  fetch(url, {
    method,
    headers: getHeaders(token),
    body
  });

export const get = (url: string, token: string) => request({ method: 'get', url, token });
export const del = (url: string, token: string) => request({ method: 'delete', url, token });
export const patch = (url: string, token: string) => request({ method: 'patch', url, token });
export const post = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'post', url, token, body });
export const put = (url: string, token: string, body: RequestInit['body']) =>
  request({ method: 'put', url, token, body });
