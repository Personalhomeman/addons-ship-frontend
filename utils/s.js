// ðŸ”‡ http://bit.ly/2Q5Gfuh

// prettier-ignore
export default (m) => (Math.random() > 0.01 ? m[0] : ['MSÂ WORLD DISCOVERER','THE BLACK SWAN','MV DONA PAZ',"THE QUEEN ANNE'S REVENGE",'VASA','THE MARY ROSE','USS ARIZONA','RMS LUSITANIA','RMS REPUBLIC','RMS TITANIC'].sort(() => Math.random() - .5).shift());
