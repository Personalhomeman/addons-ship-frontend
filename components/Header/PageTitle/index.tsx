import { Flex, Icon, Text, TypeIconName, TypeTextSize } from '@bitrise/bitkit';

import { mediaQuery } from '@/utils/media';

export type Props = {
  type: string;
  title: string;
  smaller?: boolean;
};

export default ({ type, title, smaller = false }: Props) => {
  const [isTablet, isDesktop] = mediaQuery('30rem', '60rem');

  let iconName: TypeIconName;
  switch (type) {
    case 'android':
      iconName = 'PlatformsAndroid';
      break;
    case 'ios':
      iconName = 'PlatformsApple';
      break;
    case 'cordova':
      iconName = 'PlatformsCordova';
      break;
    case 'fastlane':
      iconName = 'PlatformsFastlane';
      break;
    case 'flutter':
      iconName = 'PlatformsFlutter';
      break;
    case 'macos':
      iconName = 'PlatformsMacos';
      break;
    case 'ionic':
      iconName = 'PlatformsIonic';
      break;
    case 'react':
      iconName = 'PlatformsReact';
      break;
    case 'xamarin':
      iconName = 'PlatformsXamarin';
      break;
    case 'settings':
      iconName = 'Settings';
      break;
    default:
      iconName = 'Bitbot';
      break;
  }

  // I'm sorry... https://gph.is/22Wx2jP
  const fontSize = `x${6 - +!isDesktop - +smaller}` as TypeTextSize;

  return (
    <Flex
      direction="horizontal"
      alignChildrenVertical="middle"
      paddingHorizontal={isDesktop ? 'x0' : 'x4'}
      gap="x2"
      color="white"
    >
      <Icon name={iconName} size="2rem" />
      <Text size={fontSize} weight="bold" breakOn={isTablet ? 'all' : 'word'}>
        {title}
      </Text>
    </Flex>
  );
};
