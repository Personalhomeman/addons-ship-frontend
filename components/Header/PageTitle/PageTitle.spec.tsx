jest.mock('@/utils/media');

import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import { mediaQuery } from '@/utils/media';

import PageTitle from '.';

describe('PageTitle', () => {
  beforeEach(() => {
    (mediaQuery as jest.Mock).mockReturnValue([true]);
  });

  it('renders without errors', () => {
    const tree = shallowToJson(shallow(<PageTitle title="Some title" projectType="whatever" />));

    expect(tree).toMatchSnapshot();
  });

  it('renders without errors in smaller', () => {
    const tree = shallowToJson(shallow(<PageTitle title="Some title" projectType="whatever" smaller />));

    expect(tree).toMatchSnapshot();
  });

  it('renders without errors on mobile', () => {
    (mediaQuery as jest.Mock).mockReturnValueOnce([false]);

    const tree = shallowToJson(shallow(<PageTitle title="Some title" projectType="whatever" />));

    expect(tree).toMatchSnapshot();
  });

  const projectTypes = ['android', 'ios', 'cordova', 'ionic', 'react', 'xamarin', 'settings'];

  projectTypes.forEach(type =>
    test(`with ${type} project type`, () => {
      const tree = shallowToJson(shallow(<PageTitle title="Some title" projectType={type} />));

      expect(tree).toMatchSnapshot();
    })
  );
});
