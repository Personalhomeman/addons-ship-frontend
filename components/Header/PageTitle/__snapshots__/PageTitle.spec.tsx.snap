// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageTitle renders without errors 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="Bitbot"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle renders without errors in smaller 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="Bitbot"
    size="2rem"
  />
  <Text
    size="x5"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle renders without errors on mobile 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x4"
>
  <Icon
    name="Bitbot"
    size="2rem"
  />
  <Text
    size="x5"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with android project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="PlatformsAndroid"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with cordova project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="PlatformsCordova"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with ionic project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="PlatformsIonic"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with ios project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="PlatformsApple"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with react project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="PlatformsReact"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with settings project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="Settings"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;

exports[`PageTitle with xamarin project type 1`] = `
<Flex
  alignChildrenVertical="middle"
  color="white"
  direction="horizontal"
  gap="x2"
  paddingHorizontal="x0"
>
  <Icon
    name="PlatformsXamarin"
    size="2rem"
  />
  <Text
    size="x6"
    weight="bold"
  >
    Some title
  </Text>
</Flex>
`;
